### PR TITLE
Add REST light platform

### PR DESCRIPTION
**Pull request by robhowlett**
**Description:**

Add support for lights using a REST API.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#10114

### CODE REVIEW

**Comment by robhowlett** [comment]
The build is constantly failing due to entries such as `{"is_on": true}` in the `{% configuration %}` section. I'm struggling to find a solution to this, but continue to look for a workaround.

File: source/_components/light.rest.markdown [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    type: string
+  supported_features_template:
+    description: 'A [template](/docs/configuration/templating/#processing-incoming-data) that determines the supported features of the light from the value returned by the GET request on the resource URL. This template should compute to an array of strings. If the value is valid JSON, it will be available in the template as the variable `value_json`. Default is {% raw %}`{{value_json.supported_features}}`{% endraw %}.'
+    required: false
+    type: string
+  effect_list_template:
+    description: 'A [template](/docs/configuration/templating/#processing-incoming-data) that determines the effect list of the light from the value returned by the GET request on the resource URL. This template should compute to an array of strings. If the value is valid JSON, it will be available in the template as the variable `value_json`. Default is {% raw %}`{{value_json.effect_list}}`{% endraw %}.'
+    required: false
+    type: string
+  supported_features:
```
Comment (tinloaf) [inline source/_components/light.rest.markdown]: I think you should mention here that this is overwritten with `supported_features_template`

Comment (robhowlett) [reply source/_components/light.rest.markdown]: Thanks @tinloaf, I've now updated this as per your recommendations (commit `0dbe10c`).

**Comment by robhowlett** [comment]
This PR is related to home-assistant/home-assistant#10114 and is waiting for it to be approved/merged.

**Pull request closed without merging**